lib/hooks/userspace/hook-grpc.js

```diff
@@ -36,13 +93,15 @@ module.exports = function(version_, agent_) {
     return {};
   }
   return {
-    'src/node/src/grpc_extension.js': {
-      patch: function(extension) {
+    'src/node/src/client.js': {
+      patch: function(client) {
         agent = agent_;
-        shimmer.wrap(extension.Call.prototype, 'startBatch', startBatchWrap);
+        shimmer.wrap(client, 'makeClientConstructor',
+            makeClientConstructorWrap);
       },
-      unpatch: function(extension) {
-        shimmer.unwrap(extension.Call.prototype, 'startBatch');
+      unpatch: function(client) {
+        shimmer.unwrap(client, 'makeClientConstructor');
+        agent_.logger.info('gRPC makeClientConstructor: unpatched');
```

Comment:
This is correct (and are the same semantics we have for redis which also has to patch client constructors). The way this can be avoided is by keeping a global list of clients we've patched to unpatch them appropriately. During the redis code review this was determined to be a memory leak because the clients could be unused by the application and we maintain a reference.